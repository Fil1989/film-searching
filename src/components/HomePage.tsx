import films from '../data/imdb.json'
import 'react-responsive-carousel/lib/styles/carousel.min.css'

const Carousel = require('react-responsive-carousel').Carousel
export const plotSizing = (title: string = '', plot: string = '') => {
  if (title.length < 50) {
    return (title + plot).length < 51
      ? plot
      : plot
          .split('')
          .slice(0, 50 - title.length)
          .join('') + '...'
  } else {
    return 'title can not be loger then 50 symbols'
  }
}

function HomePage() {
  return (
    <div aria-label="Main films">
      <Carousel
        showArrows={true}
        autoPlay={true}
        infiniteLoop={true}
        className="carousell"
      >
        <div className="slider_element">
          <img
            src={films[0].poster}
            alt={films[0].title}
            className="moovie_poster"
          />
          <div
            className="slider_element_text"
            aria-label="Description of the moovie"
          >
            <h3>{films[0].title}</h3>
            <p>{plotSizing(films[0].title, films[0].plot)}</p>
          </div>
        </div>
        <div className="slider_element">
          <img
            src={films[1].poster}
            alt={films[1].title}
            className="moovie_poster"
          />
          <div
            className="slider_element_text"
            aria-label="Description of the moovie"
          >
            <h3>{films[1].title}</h3>
            <p>{plotSizing(films[1].title, films[1].plot)}</p>
          </div>{' '}
        </div>
        <div className="slider_element">
          <img
            src={films[2].poster}
            alt={films[2].title}
            className="moovie_poster"
          />
          <div
            className="slider_element_text"
            aria-label="Description of the moovie"
          >
            <h3>{films[2].title}</h3>
            <p>{plotSizing(films[2].title, films[2].plot)}</p>
          </div>{' '}
        </div>
      </Carousel>
    </div>
  )
}

export default HomePage
