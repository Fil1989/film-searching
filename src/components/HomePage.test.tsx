import { render } from '@testing-library/react'
import HomePage, { plotSizing } from './HomePage'

describe('Homepage component test', () => {
  it('Snapshot test', () => {
    const { container } = render(<HomePage />)
    expect(container).toMatchSnapshot()
  })
})
describe('Plot sizing function test', () => {
  it('not null test', () => {
    expect(plotSizing('World')).not.toBeNull()
    expect(plotSizing()).not.toBeNull()
  })
  it('title>50 symbols test', () => {
    expect(
      plotSizing(
        'Worldwide it is very popular. And who knows what will happen in future',
        'one world',
      ),
    ).toBe('title can not be loger then 50 symbols')
  })
})
