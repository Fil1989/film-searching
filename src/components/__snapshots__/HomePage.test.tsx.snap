// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component test Snapshot test 1`] = `
<div>
  <div
    aria-label="Main films"
  >
    <div
      class="carousel-root carousell"
    >
      <div
        class="carousel carousel-slider"
        style="width: 100%;"
      >
        <ul
          class="control-dots"
        >
          <li
            aria-label="slide item 1"
            class="dot selected"
            role="button"
            tabindex="0"
            value="0"
          />
          <li
            aria-label="slide item 2"
            class="dot"
            role="button"
            tabindex="0"
            value="1"
          />
          <li
            aria-label="slide item 3"
            class="dot"
            role="button"
            tabindex="0"
            value="2"
          />
        </ul>
        <button
          aria-label="previous slide / item"
          class="control-arrow control-prev"
          type="button"
        />
        <div
          class="slider-wrapper axis-horizontal"
        >
          <ul
            class="slider animated"
            style="transform: translate3d(-100%,0,0); transition-duration: 350ms;"
          >
            <li
              class="slide"
            >
              <div
                class="slider_element"
              >
                <img
                  alt="The Winds of Winter"
                  class="moovie_poster"
                  src="https://m.media-amazon.com/images/M/MV5BMTc0ZjQ5ODctYzRmNC00ZjQ3LWFlZDctN2NjNjY1MTliMjU3L2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjkxMjM5Nzc@._V1_SX300.jpg"
                />
                <div
                  aria-label="Description of the moovie"
                  class="slider_element_text"
                >
                  <h3>
                    The Winds of Winter
                  </h3>
                  <p>
                    Cersei and Loras Tyrell stand t...
                  </p>
                </div>
                 
              </div>
            </li>
            <li
              class="slide selected previous"
            >
              <div
                class="slider_element"
              >
                <img
                  alt="Ozymandias"
                  class="moovie_poster"
                  src="https://m.media-amazon.com/images/M/MV5BMTU1MjEwNjQ4MF5BMl5BanBnXkFtZTgwNzI2MTU5NTM@._V1_SX300.jpg"
                />
                <div
                  aria-label="Description of the moovie"
                  class="slider_element_text"
                >
                  <h3>
                    Ozymandias
                  </h3>
                  <p>
                    Walt goes on the run. Jesse is taken hos...
                  </p>
                </div>
              </div>
            </li>
            <li
              class="slide"
            >
              <div
                class="slider_element"
              >
                <img
                  alt="Battle of the Bastards"
                  class="moovie_poster"
                  src="https://m.media-amazon.com/images/M/MV5BZDFiZTNiMmItMDk5ZC00OGZmLTk3NmItODM5ZWE2YWMzZjliXkEyXkFqcGdeQXVyOTMyOTQyNzA@._V1_SX300.jpg"
                />
                <div
                  aria-label="Description of the moovie"
                  class="slider_element_text"
                >
                  <h3>
                    Battle of the Bastards
                  </h3>
                  <p>
                    Jon and Sansa face Ramsay Bo...
                  </p>
                </div>
                 
              </div>
            </li>
            <li
              class="slide"
            >
              <div
                class="slider_element"
              >
                <img
                  alt="The Winds of Winter"
                  class="moovie_poster"
                  src="https://m.media-amazon.com/images/M/MV5BMTc0ZjQ5ODctYzRmNC00ZjQ3LWFlZDctN2NjNjY1MTliMjU3L2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjkxMjM5Nzc@._V1_SX300.jpg"
                />
                <div
                  aria-label="Description of the moovie"
                  class="slider_element_text"
                >
                  <h3>
                    The Winds of Winter
                  </h3>
                  <p>
                    Cersei and Loras Tyrell stand t...
                  </p>
                </div>
                 
              </div>
            </li>
            <li
              class="slide selected previous"
            >
              <div
                class="slider_element"
              >
                <img
                  alt="Ozymandias"
                  class="moovie_poster"
                  src="https://m.media-amazon.com/images/M/MV5BMTU1MjEwNjQ4MF5BMl5BanBnXkFtZTgwNzI2MTU5NTM@._V1_SX300.jpg"
                />
                <div
                  aria-label="Description of the moovie"
                  class="slider_element_text"
                >
                  <h3>
                    Ozymandias
                  </h3>
                  <p>
                    Walt goes on the run. Jesse is taken hos...
                  </p>
                </div>
              </div>
            </li>
          </ul>
        </div>
        <button
          aria-label="next slide / item"
          class="control-arrow control-next"
          type="button"
        />
        <p
          class="carousel-status"
        >
          1 of 3
        </p>
      </div>
      <div
        class="carousel"
      >
        <div
          class="thumbs-wrapper axis-vertical"
        >
          <button
            aria-label="previous slide / item"
            class="control-arrow control-prev control-disabled"
            type="button"
          />
          <ul
            class="thumbs animated"
            style="transform: translate3d(0,0,0); transition-duration: 350ms;"
          >
            <li
              aria-label="slide item 1"
              class="thumb selected"
              role="button"
              style="width: 80px;"
              tabindex="0"
            >
              <img
                alt="Ozymandias"
                class="moovie_poster"
                src="https://m.media-amazon.com/images/M/MV5BMTU1MjEwNjQ4MF5BMl5BanBnXkFtZTgwNzI2MTU5NTM@._V1_SX300.jpg"
              />
            </li>
            <li
              aria-label="slide item 2"
              class="thumb"
              role="button"
              style="width: 80px;"
              tabindex="0"
            >
              <img
                alt="Battle of the Bastards"
                class="moovie_poster"
                src="https://m.media-amazon.com/images/M/MV5BZDFiZTNiMmItMDk5ZC00OGZmLTk3NmItODM5ZWE2YWMzZjliXkEyXkFqcGdeQXVyOTMyOTQyNzA@._V1_SX300.jpg"
              />
            </li>
            <li
              aria-label="slide item 3"
              class="thumb"
              role="button"
              style="width: 80px;"
              tabindex="0"
            >
              <img
                alt="The Winds of Winter"
                class="moovie_poster"
                src="https://m.media-amazon.com/images/M/MV5BMTc0ZjQ5ODctYzRmNC00ZjQ3LWFlZDctN2NjNjY1MTliMjU3L2ltYWdlL2ltYWdlXkEyXkFqcGdeQXVyNjkxMjM5Nzc@._V1_SX300.jpg"
              />
            </li>
          </ul>
          <button
            aria-label="next slide / item"
            class="control-arrow control-next"
            type="button"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
